src/views/tournament/TournamentDetailsView.jsx

```diff
@@ -8,36 +8,46 @@ class TournamentDetailsView extends Component {
     constructor(props) {
         super(props);
         this.state = {
-            tournament: {
-                name: "TournamentName",
-                teams: [],
-                matches: []
-            }
+            // tournamentId: 1,
+            // tournament: {
+            //     name: "TournamentName",
+            //     teams: [{key:"1", name: "The A Team"}, {key:"2", name: "The B Team"}],
+            //     // teams: [],
+            //     matches: [],
+            //     numberOfTeams: 8,
+            // }
         }
     }
 
     getTournament() {
-        const body = "query{ tournament(id: 1) {name, teams{name}, matches{teamHome {name}, teamAway {name}, date}}}";
-        const vars = "{}";
+        const body = "query q($id:ID!){tournament(id: $id){name, teams{name}, matches{teamHome {name}, teamAway {name}, date}, numberOfTeams}}";
```

Comment:
Don't we also need the team IDs and the match IDs?
If we have the IDs we can link to the team and match pages

Reply:
I now requests id's for tournament, match and teams.